src/world/generator/PopulationTask.php

```
@@ -29,130 +9,91 @@
+        $adjChunks = [];
+
+        foreach ($adjacent as $hash => $data) {
+            $adjChunks[$hash] = $data !== null
+                ? FastChunkSerializer::deserializeTerrain($data)->clearTerrainDirtyFlags()
+                : null;
+        }
+
+        self::setOrGenerate($mgr, $gen, $this->chunkX, $this->chunkZ, $center);
+
+        foreach ($adjChunks as $hash => $chunkObj) {
+            [$dx, $dz] = World::getXZ($hash);
```

PHPStan (2.1.15)

Cannot use array destructuring on null.


(offsetAccess.nonArray)